File: RailgunVB/Commands/Fun/Cat.vb
Imports Discord
Imports Discord.Commands
Imports RailgunVB.Core.Api
Imports RailgunVB.Core.Preconditions

Namespace Commands.Fun
    
    <Group("cat")>
    Public Class Cat
        Inherits ModuleBase
    
        Private ReadOnly _randomCat As RandomCat

        Public Sub New(randomCat As RandomCat)
            _randomCat = randomCat
        End Sub
        
        <Command, BotPerms(ChannelPermission.AttachFiles)>
        Public Async Function CatAsync() As Task
            await Context.Channel.SendFileAsync(await _randomCat.GetRandomCatAsync(), "CatImg.png")
        End Function
        
    End Class
    
End NameSpace

File: Railgun/Commands/Fun/Cat.cs
using System.Threading.Tasks;
using Discord;
using Finite.Commands;
using Railgun.Core.Api;
using Railgun.Core.Commands;
using Railgun.Core.Commands.Attributes;

namespace Railgun.Commands.Fun
{
    [Alias("cat")]
    public class Cat : SystemBase
    {
        private readonly RandomCat _randomCat;

        public Cat(RandomCat randomCat) => _randomCat = randomCat;
        
        [Command, BotPerms(ChannelPermission.AttachFiles)]
        public async Task CatAsync() 
            => await Context.Channel.SendFileAsync(await _randomCat.GetRandomCatAsync(), "CatImg.png");
    }
}